python/GafferUI/ShufflePlugValueWidget.py

```diff
@@ -0,0 +1,220 @@
+
+	def _updateFromPlug( self ) :
+
+		with self.getContext() :
+			enabled = self.getPlug()["enabled"].getValue()
+
+		for i in ( 0, 2, 3 ) :
+			self.__row[i].setEnabled( enabled )
+
+def __deletePlug( plug ) :
+
+	with Gaffer.UndoScope( plug.ancestor( Gaffer.ScriptNode ) ) :
+		plug.parent().removeChild( plug )
+
+def __plugPopupMenu( menuDefinition, plugValueWidget ):
+
+	plug = plugValueWidget.getPlug()
+	if not isinstance( plug, Gaffer.ShufflePlug ) :
+		plug = plug.ancestor( Gaffer.ShufflePlug )
+
+	if not isinstance( plug, Gaffer.ShufflePlug ):
+		return
+
+	menuDefinition.append( "/DeleteDivider", { "divider" : True } )
+	menuDefinition.append(
+		"/Delete",
+		{
+			"command" : functools.partial( __deletePlug, plug ),
+			"active" : not plugValueWidget.getReadOnly() and not Gaffer.MetadataAlgo.readOnly( plug.parent() )
+		}
+	)
+
+GafferUI.PlugValueWidget.popupMenuSignal().connect( __plugPopupMenu, scoped = False )
+
+GafferUI.PlugValueWidget.registerType( Gaffer.ShufflePlug, ShufflePlugValueWidget )
+
+##########################################################################
+# ShufflePlug Metadata
+##########################################################################
+
+Gaffer.Metadata.registerValue( Gaffer.ShufflePlug, "source", "description", "The name(s) of the source data to be shuffled. Accepts standard matching syntax (eg \"a*b\")." )
+Gaffer.Metadata.registerValue( Gaffer.ShufflePlug,
+	"destination",
+	"description",
+	"""
+	The name(s) of the destination data to be created. Use \"${source}\" to insert the name of the source data.\n
+	For example, to prefix all data with \"user:\" set the destination to \"user:${source}\").
+	"""
+)
+Gaffer.Metadata.registerValue( Gaffer.ShufflePlug, "deleteSource", "description", "Enable to delete the source data after shuffling to the destination(s)." )
+Gaffer.Metadata.registerValue( Gaffer.ShufflePlug, "enabled", "description", "Used to enable/disable this shuffle operation." )
+Gaffer.Metadata.registerValue( Gaffer.ShufflePlug, "nodule:type", "" )
+Gaffer.Metadata.registerValue( Gaffer.ShufflePlug, "*", "nodule:type", "" )
+
+##########################################################################
+# ShufflesPlug Widget
+##########################################################################
+
+## \todo: Much of this widget is reused in several places (eg CompoundDataPlugValueWidget, CameraTweaksUI, NameSwitchUI, etc).
+## Can we add more flexibility to LayoutPlugValueWidget (ie. header and footer options) and remove these other widgets?
+class ShufflesPlugValueWidget( GafferUI.PlugValueWidget ) :
+
+	def __init__( self, plug ) :
+
+		column = GafferUI.ListContainer( spacing = 4 )
+		GafferUI.PlugValueWidget.__init__( self, column, plug )
+
+		with column :
+			with GafferUI.ListContainer( GafferUI.ListContainer.Orientation.Horizontal, spacing = 4 ) :
+				GafferUI.Label( "<h3><b>Source</b></h3>" )._qtWidget().setFixedWidth( GafferUI.PlugWidget.labelWidth() )
```

Comment:
I'll defer to our resident UI guru @themissingcow on this, but to my eye these headings have far too much weight compared to the surrounding UI.

Reply:
Lowered the weight in b356807, but let me know if you'd like further changes.